src/main/scala/mesosphere/marathon/core/appinfo/impl/DefaultInfoService.scala

```diff
@@ -30,6 +30,13 @@ private[appinfo] class DefaultInfoService(
       }
     }
 
+  override def selectPodStatuses(ids: Set[PathId], selector: PodSelector): Future[Seq[PodStatus]] = {
+    val baseData = newBaseData()
+
+    val pods = ids.toVector.flatMap(groupManager.pod(_)).filter(selector.matches)
```

Comment:
before we tuned the max parallelism explicitly, how is it bounded here? So that we don't have too much parallelism in place for clusters with a lot of pods...

Reply:
We use a fixed size threadpool. We do allocate all futures but they won't steal threads from the main fork join pool